src/Templates/Microsoft.CodeAnalysis.SourceGeneratorTemplate/MyProjectGenerator.cs

```diff
@@ -0,0 +1,49 @@
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Text;
+using System.Text;
+
+namespace MyProject
+{
+    [Generator]
+    public class MyProjectGenerator : ISourceGenerator
+    {
+        public void Execute(SourceGeneratorContext context)
+        {
+            // begin creating the source we'll inject into the users compilation
+            var sourceBuilder = new StringBuilder(@"
+using System;
+namespace HelloWorldGenerated
+{
+    public static class HelloWorld
+    {
+        public static void SayHello() 
+        {
+            Console.WriteLine(""Hello from generated code!"");
+            Console.WriteLine(""The following syntax trees existed in the compilation that created this program:"");
+");
+
+            // using the context, get a list of syntax trees in the users compilation
+            var syntaxTrees = context.Compilation.SyntaxTrees;
+
+            // add the filepath of each tree to the class we're building
+            foreach (SyntaxTree tree in syntaxTrees)
+            {
+                sourceBuilder.AppendLine($@"Console.WriteLine(@"" - {tree.FilePath}"");");
```

Comment:
Consider adding 3-level indentation (12 spaces) so that the generated file is properly formatted.

```suggestion
                sourceBuilder.AppendLine($@"            Console.WriteLine(@"" - {tree.FilePath}"");");
```